lib/TemplatedPathPlugin.js

```diff
@@ -75,130 +90,205 @@ const getReplacer = (value, allowEmpty) => {
  */
 const replacePathVariables = (path, data) => {
 	const chunkGraph = data.chunkGraph;
-	const chunk = data.chunk;
-	const chunkId = chunk && chunk.id;
-	const chunkName = chunk && (chunk.name || chunk.id);
-	const chunkHash = chunk && (chunk.renderedHash || chunk.hash);
-	const chunkHashWithLength =
-		chunk && "hashWithLength" in chunk && chunk.hashWithLength;
-	const contentHashType = data.contentHashType;
-	const contentHash =
-		data.contentHash ||
-		(chunk &&
-			contentHashType &&
-			chunk.contentHash &&
-			chunk.contentHash[contentHashType]);
-	const contentHashWithLength =
-		data.contentHashWithLength ||
-		(chunk && "contentHashWithLength" in chunk
-			? chunk.contentHashWithLength[contentHashType]
-			: undefined);
-	const module = data.module;
-	const moduleId =
-		module &&
-		(module instanceof Module ? chunkGraph.getModuleId(module) : module.id);
-	const moduleHash =
-		module &&
-		(module instanceof Module
-			? chunkGraph.getRenderedModuleHash(module)
-			: module.renderedHash || module.hash);
-	const moduleHashWithLength =
-		module && "hashWithLength" in module && module.hashWithLength;
-
-	if (typeof path === "function") {
-		path = path(data);
+
+	let file = {};
+
+	if (data.filename) {
+		if (typeof data.filename === "string") {
+			const idx = data.filename.indexOf("?");
+
+			if (idx >= 0) {
+				file.path = data.filename.substr(0, idx);
+				file.query = data.filename.substr(idx);
+			} else {
+				file.path = data.filename;
+				file.query = "";
+			}
+
+			file.ext = extname(file.path);
+			file.base = basename(file.path);
+			file.name = file.base.replace(file.ext, "");
```

Comment:
For consistency (also with loader-utils) this should be splitted this way for both module and chunk:

`/some/path/file.js?query`

= `[file][query]`
= `[path][filebase][query]`
= `[path][name][ext][query]` 

| placeholder | value |
|---|---|
| [file] | /some/path/file.js |
| [query] | ?query |
| [filebase] | file.js |
| [path] | /some/path/ |
| [name] | file |
| [ext] | .js |